### PR TITLE
A real test!

### PR DESCRIPTION
**Pull request by krisalyssa**

### CODE REVIEW

File: lib/nectar/worker.ex [hunk 1]
```diff
@@ -56,20 +58,23 @@ defmodule Nectar.Worker do
     do: build_response(200, "OK", "Hello, world!") |> send_response(client)
 
   defp build_response(status_code, status_message, message_body) do
+    # account for the "\r\n" pair at the end of the message body
+    content_length = byte_size(message_body) + 2
```
Comment (jeffutter) [inline lib/nectar/worker.ex:62]: Is it part of the spec that the last line ends with that? (asking because I don't know)

Comment (krisalyssa) [reply lib/nectar/worker.ex:62]: I don't know either. Calling Nectar from HTTPoison without this nets a `:bad_request` error from hackney. This cleared up that error.

I suppose somebody should go read the spec some time soon....

Comment (jeffutter) [reply lib/nectar/worker.ex:62]: ¯\_(ツ)_/¯ Ok, I'll go with it for now until we know better.

**Review by jeffutter — APPROVED** [review]

File: lib/nectar/worker.ex [hunk 2]
```diff
@@ -56,20 +58,23 @@ defmodule Nectar.Worker do
...
     Date: #{get_datetime()}
 
-    #{message_body}
+    #{message_body}\r
     """
-    |> String.trim()
   end
 
   defp send_response(response, client) do
+    Logger.debug(">>>\n#{response}<<<")
```
Comment (WizardOfOgz) [inline lib/nectar/worker.ex:77]: @CraigCottingham 

```
Logger.debug fn -> 
  ">>>\n#{response}<<<"
end
```

Avoids performing the string interpolation unless the logging level is actually set to debug.

Comment (krisalyssa) [reply lib/nectar/worker.ex:77]: `credo` caught that, too. Already fixed in the next go-round.

File: test/nectar_test.exs [hunk 3]
```diff
@@ -2,7 +2,11 @@ defmodule NectarTest do
   use ExUnit.Case
   doctest Nectar
 
+  test "a known good HTTP server" do
+    assert {:ok, %HTTPoison.Response{}} = HTTPoison.get("http://elixir-lang.org/")
```
Comment (WizardOfOgz) [inline test/nectar_test.exs:6]: This is unlikely to break, but having a test like this is generally discouraged. It does not test anything in Nectar. The only things it tests are HTTPoison and the responsiveness of elixir-lang.org. I'd recommend removing this altogether.

**Pull request merged**